### ISSUE TITLE
Update small (<2B) model configuration for LUMI

### ISSUE THREAD

**Issue by spyysalo** [OpenEuroLLM/Taskboard#16]
The ~1.7B param model configuration https://github.com/spyysalo/lumi-fineweb-replication/ is suboptimal in a number of ways. It should be updated to more closely match the Open-Sci configuration.

**Comment by spyysalo** [comment]
Completed an experiment training 1.7B param models to 30BT on FineWeb-edu, one with the original cosine LR schedule and another with a WSD schedule. Both loss curves and downstream evaluations indicate the WSD is better 🙌 

<img width="607" alt="Image" src="https://github.com/user-attachments/assets/0b2f32b7-bd44-4149-80f2-5cb2e9dae3c3" />
<img width="657" alt="Image" src="https://github.com/user-attachments/assets/0bb8a4fb-0fa0-403e-a873-d7492fa9777f" />
<img width="399" alt="Image" src="https://github.com/user-attachments/assets/a11f79f6-e9c5-4e27-a1ca-fb553de75a7f" />

**Comment by spyysalo** [comment]
Same for a 100BT run. Differences at 100BT are less pronounced but WSD does still show lower loss and higher downstream performance at the last checkpoint.

<img width="611" alt="Image" src="https://github.com/user-attachments/assets/e18df8d5-0ee0-40a1-8658-cfa10ddb253e" />
<img width="662" alt="Image" src="https://github.com/user-attachments/assets/9e001228-057e-424b-8db7-1dbe0d0a52a0" />

**Comment by spyysalo** [comment]
Increasing LR from 3e-4 to 4e-3 without other changes results in early deconvergence, but after adding in `--qk-layernorm` the process is stable and results in a lower training loss on a 30BT FW-edu training run.

<img width="602" alt="Image" src="https://github.com/user-attachments/assets/d376090e-67cd-4881-9c59-ba1a3d32f099" />
<img width="602" alt="Image" src="https://github.com/user-attachments/assets/cccee98e-a250-45dd-babd-054c63015ca0" />

**Comment by spyysalo** [comment]
As there doesn't appear to be any off-the-shelf HF transformers model that we could convert into with QK norm using RMSNorm, I'll set aside `--qk-layernorm` for the moment

**Comment by spyysalo** [comment]
Naively changing seq len from 2K to 4K and hence GBS from 2M to 4M results in the expected increase in loss as well as a very substantial drop in eval results.

<img width="602" alt="Image" src="https://github.com/user-attachments/assets/9e7a2932-c8b3-4f84-a6ab-edd6aa1df3d8" />
<img width="659" alt="Image" src="https://github.com/user-attachments/assets/33d4b6b6-1271-45df-b72d-8d7b799fc1ba" />

**Comment by spyysalo** [comment]
Increasing LR for the 4K seq len/4M token GBS setup broadly matches the loss of the 2K/2M (LR 3e-4) at 1e-3 and achieves lower loss with 2e-3 and 3e-3 on a 30BT training run on FineWeb-edu. Higher LRs appear unstable (no `--qk-layernorm`). Downstream eval results appear to only catch up at 3e-3 in the same runs. The LR 3e-3 run shows some loss spikes.

(Additional confound when comparing the 2K/2M and 4K/4M configurations: as warmup is currently defined in terms of iterations, the 4K/4M also has double the warmup.)

<img width="659" alt="Image" src="https://github.com/user-attachments/assets/14759625-6ba4-4c78-883c-e79c8895f0f0" />
<img width="659" alt="Image" src="https://github.com/user-attachments/assets/df7c0bb2-a9de-444e-bfb3-ed1b509c9736" />
<img width="659" alt="Image" src="https://github.com/user-attachments/assets/86cdcffd-198f-4c68-8a2f-b081f92d00a4" />

**Comment by spyysalo** [comment]
Increasing warmup from 500 steps to 1000 or 1430 (10% of the 30BT token run steps) shows the expected initial difference in loss, with near-identical loss after approx. 10B tokens

<img width="651" alt="Image" src="https://github.com/user-attachments/assets/a68d1559-3c63-4bc3-9d80-dd6f841d6ad6" />

Downstream evaluation results also suggest that the 500 step warmup may yield better performance very early on in training, but the ordering is inverted around 10B tokens. While the differences are not great, the 10% warmup variant shows the best performance from around 15B tokens to the end of training at 30B tokens.

<img width="651" alt="Image" src="https://github.com/user-attachments/assets/b8125581-76c4-41cf-a8b5-6a2012b55e4f" />

**Comment by spyysalo** [comment]
Increasing the seq len from 2K to 4K while halving the GBS (thus keeping the global batch size equal in terms of tokens) gives effectively identical loss curves

<img width="651" alt="Image" src="https://github.com/user-attachments/assets/dd3f5ac2-614c-49bc-a634-cc025fc18bd5" />

... but somewhat surprisingly appears to result in consistently lower evaluation performance

<img width="651" alt="Image" src="https://github.com/user-attachments/assets/fe666497-a6a2-4f57-9cdd-92daca9a335a" />

the difference appears to remain as warmup is increased

<img width="651" alt="Image" src="https://github.com/user-attachments/assets/16b1e945-1a5a-447d-aec4-3ea885b92bd0" />

**Comment by spyysalo** [comment]
It's likely that the apparent drop in performance when moving to a 4K seq length could be reduced or eliminated by reoptimizing other hyperparameters. (A quick experiment increasing the rope theta to 20K did not produce a noticeable difference in performance.) While a longer seq length is desirable in general, it may not be necessary for small models used to compare dataset quality etc.

**Comment by spyysalo** [comment]
I've committed the updated configuration to https://github.com/OpenEuroLLM/task4.3/tree/main/scripts/lumi . This should still be verified in a longer run (at least 100BT), but I'm tentatively calling the update complete. In partial summary of the above, the following changes to align more closely with the Open-Sci configuration were evaluated and included in the update as either improving or at least not negatively impacting performance when training on FineWeb-edu and evaluating on ARC (easy and challenge), PIQA, HellaSwag, and OpenBookQA:

* Warmup-Stable-Decay learning rate schedule with 20% cooldown
* Extended warmup, from a fixed 500 iterations to 10% (e.g. for a 30BT run approx 1500 iterations)
* Lower weight decay, from 0.1 to 0.05 (Megatron-LM default is 0.01)

Additionally the following settings not related to the model configuration were changed from the https://github.com/spyysalo/lumi-fineweb-replication `train-gpt.sh` script:

* Default node count, slurm account, tokenizer (Gemma 3), data, train token count and save interval
* Use `--train-iters`, `--lr-warmup-iters` etc. arguments instead of `--train-samples` etc.
* Set `--distributed-timeout-minutes 10` instead of `30`

The following were considered but not adopted (see above for details):

* Sequence length 4096 instead of 2048
* Global batch size of ~4M tokens instead of ~2M tokens

**Comment by spyysalo** [comment]
In a 100BT training run on FineWeb-edu with the Gemma 3 tokenizer, the updated version https://github.com/OpenEuroLLM/task4.3/tree/main/scripts/lumi shows the characteristic WSD loss pattern and achieves lower final loss than the original "FineWeb replication" setup.

<img width="609" alt="Image" src="https://github.com/user-attachments/assets/a0a46512-1329-414a-ae36-38445067e21a" />

The pattern where the updated version shows lower performance early on in training but catches up later is seen also in downstream evaluation results. While the differences over most of the training process are small, the updated version does also show slightly higher average performance for these tasks in the final stages of training.

<img width="655" alt="Image" src="https://github.com/user-attachments/assets/cfa7bad9-71fa-4fa6-a927-e8c69d9a73b3" />

There is no indication that the updated version would be worse than the original, and the loss curve in particular suggests it is an improvement. Given also the other advantages of the WSD schedule, I'm happy to endorse moving forward with this version for the following series of experiments.

**Comment by JeniaJitsev** [comment]
It would be good to train FineWeb on 300B and have direct comparison to open-sci-ref numbers to check everything is okay. 30B and 100B we dont have a reference comparison to.

**Comment by JeniaJitsev** [comment]
> Increasing the seq len from 2K to 4K while halving the GBS (thus keeping the global batch size equal in terms of tokens) gives effectively identical loss curves
> 
> <img alt="Image" width="651" src="https://private-user-images.githubusercontent.com/587850/439833204-dd3f5ac2-614c-49bc-a634-cc025fc18bd5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYzMDEzMzYsIm5iZiI6MTc1NjMwMTAzNiwicGF0aCI6Ii81ODc4NTAvNDM5ODMzMjA0LWRkM2Y1YWMyLTYxNGMtNDliYy1hNjM0LWNjMDI1ZmMxOGJkNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyN1QxMzIzNTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NWU1NjJkMjc5ZjNiMmIxMmE2ODUyNzYzYzk0Y2NjM2EwNWVmMDYwMjNhMTJiY2E4YzJiNzZhODQ3Njg5NTBiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.G77RGo6OXH5lInRoukLGXzskq7rf9M4WeMmmJQZsp1E">
> ... but somewhat surprisingly appears to result in consistently lower evaluation performance
> 
> <img alt="Image" width="651" src="https://private-user-images.githubusercontent.com/587850/439833453-fe666497-a6a2-4f57-9cdd-92daca9a335a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYzMDEzMzYsIm5iZiI6MTc1NjMwMTAzNiwicGF0aCI6Ii81ODc4NTAvNDM5ODMzNDUzLWZlNjY2NDk3LWE2YTItNGY1Ny05Y2RkLTkyZGFjYTlhMzM1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyN1QxMzIzNTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNWNiMTExZjNlNGY3ZmM1YjkyZTJhZDUwYWNjZmNkZmJhNzk2ODJlNmIyOTEyYTA0ZGNjZGZjZGQ3MTFjMTBiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.01wHUXQFKomHeh0kgWz6LxeDMZUrnlDkXcKL-abu3Y8">
> the difference appears to remain as warmup is increased
> 
> <img alt="Image" width="651" src="https://private-user-images.githubusercontent.com/587850/439833768-16b1e945-1a5a-447d-aec4-3ea885b92bd0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYzMDEzMzYsIm5iZiI6MTc1NjMwMTAzNiwicGF0aCI6Ii81ODc4NTAvNDM5ODMzNzY4LTE2YjFlOTQ1LTFhNWEtNDQ3ZC1hZWM0LTNlYTg4NWI5MmJkMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyN1QxMzIzNTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNDAxZTEzZTgyODc1YmZlNDY2OTQ2MDhmNDhkZmQwNjE5M2I0OTZhZGMxODYzOTU0MTMxYWEwYTBjNmYxODc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.fxlm3eVLJaMBawfJvKIW9XSTVutKXt76hkbLZUM9pJc">


I think that is a nice reminder that one should not rely on test loss only and **always** also check downstream evals.
